sample/samplers.go

```diff
@@ -90,8 +90,7 @@ func (s *Sampler) sample(tokens []token) (token, error) {
 		sortLogits(tokens)
 	}
 
-	tokens = temperature(tokens, s.temperature)
-	tokens = softmax(tokens)
+	tokens = temperatureAndSoftmax(tokens, s.temperature)
```

Comment:
Do we want to keep calling this `temperature`? Is there a case where we we wouldn't want softmax?

Reply:
should be okay to call temperature as that's the only param needed. need softmax in every case as the probabilities are needed for all subsequent steps

Reply:
I think we can name it `temperature` then for simplicity